File: packages/@ros-cdk/core/lib/ros-pseudo.ts
import { Token } from "./token";

const ROS_REGION = "ALIYUN::Region";
const ROS_STACKID = "ALIYUN::StackId";
const ROS_STACKNAME = "ALIYUN::StackName";
const ROS_NOVALUE = "ALIYUN::NoValue";
const ROS_ACCOUNTID = "ALIYUN::AccountId";
const ROS_TENANTID = "ALIYUN::TenantId";

/**
 * Accessor for pseudo parameters
 */
export class RosPseudo {
  private constructor() {}

  public static get accountId(): string {
    return this.asString(ROS_ACCOUNTID);
  }

  public static get region(): string {
    return this.asString(ROS_REGION);
  }

  public static get stackId(): string {
    return this.asString(ROS_STACKID);
  }

  public static get stackName(): string {
    return this.asString(ROS_STACKNAME);
  }

  public static get noValue(): string {
    return this.asString(ROS_NOVALUE);
  }

  public static get tenantId(): string {
    return this.asString(ROS_TENANTID);
  }

  private static asString(name: string) {
    return Token.asString({ Ref: name }, { displayHint: name });
  }
}

File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Core/AlibabaCloud/SDK/ROS/CDK/Core/RosPseudo.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Core
{
    /// <summary>Accessor for pseudo parameters.</summary>
    [JsiiClass(nativeType: typeof(AlibabaCloud.SDK.ROS.CDK.Core.RosPseudo), fullyQualifiedName: "@alicloud/ros-cdk-core.RosPseudo")]
    public class RosPseudo : DeputyBase
    {
        /// <summary>Used by jsii to construct an instance of this class from a Javascript-owned object reference</summary>
        /// <param name="reference">The Javascript-owned object reference</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected RosPseudo(ByRefValue reference): base(reference)
        {
        }

        /// <summary>Used by jsii to construct an instance of this class from DeputyProps</summary>
        /// <param name="props">The deputy props</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected RosPseudo(DeputyProps props): base(props)
        {
        }

        [JsiiProperty(name: "accountId", typeJson: "{\"primitive\":\"string\"}")]
        public static string AccountId
        {
            get => GetStaticProperty<string>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.RosPseudo))!;
        }

        [JsiiProperty(name: "noValue", typeJson: "{\"primitive\":\"string\"}")]
        public static string NoValue
        {
            get => GetStaticProperty<string>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.RosPseudo))!;
        }

        [JsiiProperty(name: "region", typeJson: "{\"primitive\":\"string\"}")]
        public static string Region
        {
            get => GetStaticProperty<string>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.RosPseudo))!;
        }

        [JsiiProperty(name: "stackId", typeJson: "{\"primitive\":\"string\"}")]
        public static string StackId
        {
            get => GetStaticProperty<string>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.RosPseudo))!;
        }

        [JsiiProperty(name: "stackName", typeJson: "{\"primitive\":\"string\"}")]
        public static string StackName
        {
            get => GetStaticProperty<string>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.RosPseudo))!;
        }

        [JsiiProperty(name: "tenantId", typeJson: "{\"primitive\":\"string\"}")]
        public static string TenantId
        {
            get => GetStaticProperty<string>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.RosPseudo))!;
        }
    }
}
